vespajlib/src/main/java/com/yahoo/slime/ObjectInserter.java

```diff
@@ -1,25 +1,73 @@
 // Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.slime;
 
+import java.util.Optional;
+
 /**
  * Helper class for inserting values into an ObjectValue.
  * For justification read Inserter documentation.
  **/
-final class ObjectInserter implements Inserter {
+public final class ObjectInserter implements Inserter {
```

Comment:
I suggest renaming ObjectInserter to ObjectSymbolInserter (using symbol values) and to extract the JsonObjectInserter from the json parser into an ObjectInserter class (using field names). This will match better with the C++ implementation and will also avoid object duality.